### PR TITLE
Tunnel log entries were not decoded correctly

### PR DESCRIPTION
**Pull request by mkestler-mozilla**
On windows, the conversion from the fixed length buffer read from the tunnel's ringbuffer resulted in a 512b fixed length string being written to the log.  When encoded and sent to guardian for a support request that turned in to 512 /u0000s which caused it to be rejected by FxA for length.
I now truncate the buffer before conversion.  Also fixed a casting issue that was causing a warning.

### CODE REVIEW

File: src/platforms/windows/daemon/windowstunnellogger.cpp [hunk 1]
```diff
@@ -107,6 +108,10 @@ void WindowsTunnelLogger::process(int index) {
 
   QByteArray message((const char*)data + RINGLOG_TIMESTAMP_SIZE,
                      RINGLOG_MESSAGE_SIZE);
+  int nullIndex = message.indexOf(0x0);
```
Comment (oskirby) [inline src/platforms/windows/daemon/windowstunnellogger.cpp:111]: Oooh! Good catch! We could also have used `QString::fromUtf(const char *str, int size = -1)` which would simplify this even further.

File: src/platforms/windows/daemon/windowstunnellogger.cpp [hunk 2]
```diff
@@ -107,6 +108,10 @@ void WindowsTunnelLogger::process(int index) {
...
+  if (nullIndex > 0) {
```
Comment (bakulf) [inline src/platforms/windows/daemon/windowstunnellogger.cpp:112]: `>=` ?

Comment (mkestler-mozilla) [reply src/platforms/windows/daemon/windowstunnellogger.cpp:112]: No, Owen's suggestion is better.  The true root is trying to decode UTF16 as UTF8.  I'm just verifying that now.

File: src/platforms/windows/daemon/windowstunnellogger.cpp (hunk 1 above)
Comment (mkestler-mozilla) [reply src/platforms/windows/daemon/windowstunnellogger.cpp:111]: Correct.  I'll post an update doing the proper conversion.

Comment (mkestler-mozilla) [reply src/platforms/windows/daemon/windowstunnellogger.cpp:110]: Turns out it isn't.  Left in the current truncation.

File: src/platforms/windows/daemon/windowstunnellogger.cpp (hunk 2 above)
Comment (mkestler-mozilla) [reply src/platforms/windows/daemon/windowstunnellogger.cpp]: Conversion didn't work, so I made this change.

**Review by bakulf — APPROVED** [review]

**Pull request merged**